packages/web-ui/src/sections/Document/Sidebar/Files/index.tsx

```diff
@@ -276,41 +277,100 @@ function FileNode({
   )
 }
 
+type DeletableType = 'file' | 'folder'
+type DeletableElement<T extends DeletableType> = T extends 'file'
+  ? {
+      type: T
+      documentUuid: string
+      name: string
+    }
+  : {
+      type: T
+      path: string
+      name: string
+    }
+
 export function FilesTree({
   currentPath,
   documents,
   navigateToDocument,
   createFile,
   destroyFile,
+  destroyFolder,
+  isDestroying,
 }: {
   createFile: (args: { path: string }) => Promise<void>
   destroyFile: (documentUuid: string) => Promise<void>
+  destroyFolder: (path: string) => Promise<void>
   documents: SidebarDocument[]
   currentPath: string | undefined
   navigateToDocument: (documentUuid: string) => void
+  isDestroying: boolean
 }) {
   const togglePath = useOpenPaths((state) => state.togglePath)
   const rootNode = useTree({ documents })
+  const [deletableNode, setDeletable] =
+    useState<DeletableElement<DeletableType> | null>(null)
 
   useEffect(() => {
     if (currentPath) {
       togglePath(currentPath)
     }
   }, [currentPath, togglePath])
+  const onConfirmDelete = useCallback(
+    async <T extends DeletableType>(deletable: DeletableElement<T>) => {
+      if (deletable.type === 'file') {
+        await destroyFile(deletable.documentUuid)
+      } else if (deletable.type === 'folder') {
+        await destroyFolder(deletable.path)
+      }
 
+      setDeletable(null)
+    },
+    [destroyFile, destroyFolder, deletableNode, setDeletable],
+  )
+
+  const deletingFolder = deletableNode?.type === 'folder'
   return (
-    <FileTreeProvider
-      currentPath={currentPath}
-      onNavigateToDocument={navigateToDocument}
-      onCreateFile={async (path) => {
-        createFile({ path })
-      }}
-      onDeleteFile={destroyFile}
-      onDeleteFolder={async (path) => {
-        console.log('onDeleteFolder', path)
-      }}
-    >
-      <FileNode node={rootNode} />
-    </FileTreeProvider>
+    <>
+      <FileTreeProvider
+        currentPath={currentPath}
+        onNavigateToDocument={navigateToDocument}
+        onCreateFile={async (path) => {
+          createFile({ path })
+        }}
+        onDeleteFile={({ node, documentUuid }) => {
+          setDeletable({ type: 'file', documentUuid, name: node.name })
```

Comment:
ditto, don't let ferran fool you, enums are good for u

Reply:
lol